framework/src/meshmodifiers/ElementSubdomainModifierBase.C

```diff
@@ -0,0 +1,662 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementSubdomainModifierBase.h"
+#include "DisplacedProblem.h"
+
+#include "libmesh/parallel_algebra.h"
+#include "libmesh/parallel.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/remote_elem.h"
+#include "libmesh/parallel_ghost_sync.h"
+#include "libmesh/petsc_vector.h"
+
+InputParameters
+ElementSubdomainModifierBase::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
+
+  // ESMs only operate on the undisplaced mesh to gather subdomain and sideset information. This
+  // information is used to modify both the undisplaced and the displaced meshes. It is the
+  // developer's responsibility to make sure the element IDs and sideset info are consistent across
+  // both meshes.
+  params.set<bool>("use_displaced_mesh") = false;
+  params.suppressParameter<bool>("use_displaced_mesh");
+
+  params.addParam<BoundaryName>(
```

Comment:
```suggestion
  params.addDeprecatedParam<BoundaryName>(
```

Reply:
will remove it from auto-suggest in vscode plugin

Reply:
Hi. I didn't use the `DeprecatedParam` here as these parameters (`moving_boundary_name` and `complement_moving_boundary_name`) cannot be used anymore. Instead, using these will lead to an error message explaining to the user the updated parameters. I did use `DeprecatedParam` for the `apply_initial_conditions`, since that one can still be used.

Reply:
It does not hurt to use deprecatedParam here.
and the nice thing is that it removes it from the documentation

Reply:
added